### PR TITLE
Feature/player integrations

### PR DESCRIPTION
**Pull request by maniland**
adds commercial player pages for:
- Theoplayer
- JWplayer
- Bitmovin

### CODE REVIEW

File: docs/source/integrations/Theoplayer.md [hunk 1]
```diff
@@ -0,0 +1,84 @@
+# THEOplayer
+
+[THEOplayer](https://www.theoplayer.com) is an industry leading HTML5 video player.
+
+The [vuplay-theoplayer](https://github.com/vualto/vuplay-theoplayer) repository demonstrates at a lower level how to integrate [vudrm](https://docs.vualto.com/projects/vudrm/en/latest/index.html) with THEOplayer.
+
+## Basic set up
```
Comment (lorenzo-vualto) [inline docs/source/integrations/Theoplayer.md]: Change to "setup"

File: docs/source/integrations/Theoplayer.md [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+          licenseAcquisitionURL:
+            "https://playready-license.drm.technology/rightsmanager.asmx"
+        }
+      }
+    }
+  ]
+};
+```
+
+## Fairplay example
```
Comment (lorenzo-vualto) [inline docs/source/integrations/Theoplayer.md]: I believe it is "FairPlay"

File: docs/source/integrations/bitmovin.md [hunk 3]
```diff
@@ -0,0 +1,89 @@
+# Bitmovin
+
+[Bitmovin](https://bitmovin.com) is an industry leading HTML5 video player.
+
+The [vuplay-bitmovin](https://github.com/Vualto/vuplay-bitmovin) repository demonstrates at a lower level how to integrate [vudrm](https://docs.vualto.com/projects/vudrm/en/latest/index.html) with Bitmovin.
+
+## Basic set up
```
Comment (lorenzo-vualto) [inline docs/source/integrations/bitmovin.md]: Change to "setup"

File: docs/source/integrations/bitmovin.md [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+
+```javascript
+var container = document.getElementById("my-player");
+var VUDRM_TOKEN = "<your-vudrm-token>";
```
Comment (lorenzo-vualto) [inline docs/source/integrations/bitmovin.md]: Change to `vudrmToken `

File: docs/source/integrations/bitmovin.md [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+
+## Widevine example
+
+```javascript
+source.drm = {
+  widevine: {
+    LA_URL: "https://widevine-proxy.drm.technology/proxy",
+    prepareMessage: function(keyMessage) {
+      return JSON.stringify({
+        token: VUDRM_TOKEN,
```
Comment (lorenzo-vualto) [inline docs/source/integrations/bitmovin.md]: Change to `vudrmToken `

File: docs/source/integrations/bitmovin.md [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+        drm_info: Array.apply(null, new Uint8Array(keyMessage.message)),
+        kid: "<CONTENT-KEY-ID>"
```
Comment (lorenzo-vualto) [inline docs/source/integrations/bitmovin.md]: Change to `"<your-content-key-id>"`

File: docs/source/integrations/bitmovin.md [hunk 7]
```diff
@@ -0,0 +1,89 @@
...
+  }
+};
+```
+
+## PlayReady example
+
+```javascript
+source.drm = {
+    playready: {
+        LA_URL: `https://playready-license.drm.technology/rightsmanager.asmx?token=" + encodeURIComponent(VUDRM_TOKEN);
```
Comment (lorenzo-vualto) [inline docs/source/integrations/bitmovin.md]: Invalid JS. There is a combination of string interpolation "`" and magic strings.
Copy & paste whichever you prefer 😉

```
`https://playready-license.drm.technology/rightsmanager.asmx?token=${encodeURIComponent(VUDRM_TOKEN)}`;
```

```
"https://playready-license.drm.technology/rightsmanager.asmx?token=" + encodeURIComponent(VUDRM_TOKEN);
```

File: docs/source/integrations/bitmovin.md [hunk 8]
```diff
@@ -0,0 +1,89 @@
...
+    }
+}
+```
+
+## Fairplay example
```
Comment (lorenzo-vualto) [inline docs/source/integrations/bitmovin.md]: I believe it is "FairPlay"

File: docs/source/integrations/bitmovin.md [hunk 9]
```diff
@@ -0,0 +1,89 @@
...
+
+```javascript
+source.drm = {
+  fairplay: {
+    certificateURL: "<YOUR-FAIRPLAY-CERT>",
```
Comment (lorenzo-vualto) [inline docs/source/integrations/bitmovin.md]: Change to `"<your-fairplay-cert>"`

File: docs/source/integrations/bitmovin.md (hunk 7 above)
Comment (lorenzo-vualto) [inline docs/source/integrations/bitmovin.md]: Change to `vudrmToken`

File: docs/source/integrations/bitmovin.md [hunk 10]
```diff
@@ -0,0 +1,89 @@
...
+    LA_URL: "<FAIRPLAY-LICENSE-SERVER-URL>",
```
Comment (lorenzo-vualto) [inline docs/source/integrations/bitmovin.md]: Change to `"<your-fairplay-license-server-url>"`

File: docs/source/integrations/bitmovin.md [hunk 11]
```diff
@@ -0,0 +1,89 @@
...
+    certificateHeaders: {
+      "x-vudrm-token": [VUDRM_TOKEN]
```
Comment (lorenzo-vualto) [inline docs/source/integrations/bitmovin.md]: Change to `[vudrmToken]`

File: docs/source/integrations/bitmovin.md [hunk 12]
```diff
@@ -0,0 +1,89 @@
...
+    },
+    headers: {
+      "Content-Type": "application/json"
+    },
+    prepareMessage: function(keyMessageEvent, keySession) {
+      return JSON.stringify({
+        token: VUDRM_TOKEN,
```
Comment (lorenzo-vualto) [inline docs/source/integrations/bitmovin.md]: Change to `vudrmToken`

File: docs/source/integrations/integrations.rst [hunk 13]
```diff
@@ -5,5 +5,8 @@ INTEGRATIONS
    :maxdepth: 2
    :glob:
 
-   *
+   bitmovin.md
+   jwplayer.md
+   Theoplayer.md
```
Comment (lorenzo-vualto) [inline docs/source/integrations/integrations.rst]: Inconsistent naming, capital "T"

File: docs/source/integrations/integrations.rst [hunk 14]
```diff
@@ -5,5 +5,8 @@ INTEGRATIONS
...
+   ios-fairplay-sdk.md
```
Comment (lorenzo-vualto) [inline docs/source/integrations/integrations.rst]: Inconsistent naming, kebab-case

File: docs/source/integrations/ios-fairplay-sdk.md [hunk 15]
```diff
@@ -1,138 +1,140 @@
...
-2. Navigate to your target and add vudrmFairPlay.framework to your ‘Embedded Binaries’. You should see that vudrmFairPlay.framework also now appears under ‘Linked Frameworks and Libraries’: 
-
-### INFORMATION ABOUT APPLICATION TRANSPORT SECURITY (ATS) ### 
-iOS/tvOS 9 introduces Application Transport Security (ATS) which restricts the use of insecure HTTP. 
-In order to permit playback of content over insecure HTTP exemptions need to be added into your application's Info.plist. For example to disable ATS for any connection you would add the following into your application's Info.plist: 
-<dict> 
-<key>NSAppTransportSecurity</key> <dict> 
+
+1.      Online Playback / Download and Offline Playback using a persisted token and license.
+    And legacy methods:
```
Comment (lorenzo-vualto) [inline docs/source/integrations/ios-fairplay-sdk.md]: Is this colon intended here?

File: docs/source/integrations/ios-fairplay-sdk.md [hunk 16]
```diff
@@ -1,138 +1,140 @@
...
+2.      Make use of Apple’s AVPlayer and AVPlayerViewController to present users with the platform
+    default skins, with DRM content.
+3.      Create your own video player based on AVPlayer.
+    As an evolution of the the hugely successful VUPLAY SDK, this SDK has a number of key benefits:
+    ● Complete control over the entire AVPlayer lifecycle
+    ● Numerous optimisations to enable faster playback
+    ● Bitcode support
+    ● Support for building/linking against the simulators
+    The SDK is fully supported in both Objective-C and Swift applications. A demo application written in Swift is available on request.
+    Please contact support@vualto.com to request a download of the SDK.
```
Comment (lorenzo-vualto) [inline docs/source/integrations/ios-fairplay-sdk.md]: Change to `<support@vualto.com>` to enable as an email link

File: docs/source/integrations/ios-fairplay-sdk.md [hunk 17]
```diff
@@ -1,138 +1,140 @@
...
+### XCODE INTEGRATION
+
+1. Copy the vudrmFairPlay.framework into your application's project directory
+2. Navigate to your target and add vudrmFairPlay.framework to your ‘Embedded Binaries’. You should see that vudrmFairPlay.framework also now appears under ‘Linked Frameworks and Libraries’:
+
+### INFORMATION ABOUT APPLICATION TRANSPORT SECURITY (ATS)
+
+iOS/tvOS 9 introduces Application Transport Security (ATS) which restricts the use of insecure HTTP.
+In order to permit playback of content over insecure HTTP exemptions need to be added into your application's Info.plist. For example to disable ATS for any connection you would add the following into your application's Info.plist:
+<dict>
```
Comment (lorenzo-vualto) [inline docs/source/integrations/ios-fairplay-sdk.md]: Surround snippet in triple back-ticks and apply appropriate indentation

File: docs/source/integrations/ios-fairplay-sdk.md [hunk 18]
```diff
@@ -1,138 +1,140 @@
...
+<key>NSAppTransportSecurity</key> <dict>
 <key>NSAllowsArbitraryLoads</key> <true/>
-</dict> 
-</dict> 
-More information about ATS can be found in Apple’s TechNote https://developer.apple.com/library/prerelease/ios/technotes/App-Transport-Security-Technote/ 
+</dict>
+</dict>
+More information about ATS can be found in Apple’s TechNote https://developer.apple.com/library/prerelease/ios/technotes/App-Transport-Security-Technote/
```
Comment (lorenzo-vualto) [inline docs/source/integrations/ios-fairplay-sdk.md]: Add space before and after code snippet

Comment (lorenzo-vualto) [inline docs/source/integrations/ios-fairplay-sdk.md]: Surround URL with `<...>` or create hyperlink

```
[Apple’s TechNote](https://developer.apple.com/library/prerelease/ios/technotes/App-Transport-Security-Technote/)
```

File: docs/source/integrations/ios-fairplay-sdk.md [hunk 19]
```diff
@@ -1,138 +1,140 @@
...
 The example uses a single dynamically assigned instance. It would be expected that your application will manage the applications offline assets, including their ID’s, locations and download status. This method uses AVAssetDownloadURLSession with a URLSessionConfiguration to create an AVAssetDownloadTask for each configured asset.
- 
+
 The current example initialises an AVAssetDownloadURLSession with a URLSessionConfiguration upon
 loading the ViewController:
- 
+
 backgroundConfiguration = URLSessionConfiguration.background(withIdentifier: "AssetDownloadConfigurationIdentifier")
-        	assetUrlSession = AVAssetDownloadURLSession(configuration: backgroundConfiguration!,
+assetUrlSession = AVAssetDownloadURLSession(configuration: backgroundConfiguration!,
```
Comment (lorenzo-vualto) [inline docs/source/integrations/ios-fairplay-sdk.md]: Surround code snippet in triple back-ticks

File: docs/source/integrations/ios-fairplay-sdk.md [hunk 20]
```diff
@@ -1,138 +1,140 @@
...
- 
+
 The asset configuration is then assigned upon use of the download or
 play buttons. To play an asset we create the assetDownloadTask and then assign the associated urlAsset to a player:
- 
+
 let asset = AVURLAsset(url: assetURL!)
 self.drm = vudrmFairPlay(asset: asset, assetName: assetName, token: token)
-               assetDownloadTask = assetUrlSession.makeAssetDownloadTask(asset: asset, assetTitle: assetName,
+assetDownloadTask = assetUrlSession.makeAssetDownloadTask(asset: asset, assetTitle: assetName,
```
Comment (lorenzo-vualto) [inline docs/source/integrations/ios-fairplay-sdk.md]: Surround code snippet in triple back-ticks

File: docs/source/integrations/ios-fairplay-sdk.md [hunk 21]
```diff
@@ -1,138 +1,140 @@
...
-       	present(playerViewController, animated: true) {
+let player = AVPlayer(playerItem: playerItem)
+let playerViewController = AVPlayerViewController()
+playerViewController.player = player
+present(playerViewController, animated: true) {
 player.play()
 }
 And to download, instead of assigning the asset to a player, call:
- 
+
```
Comment (lorenzo-vualto) [inline docs/source/integrations/ios-fairplay-sdk.md]: Surround code snippet in triple back-ticks

File: docs/source/integrations/ios-fairplay-sdk.md [hunk 22]
```diff
@@ -1,138 +1,140 @@
...
 assetDownloadTask.resume().
- 
+
```
Comment (lorenzo-vualto) [inline docs/source/integrations/ios-fairplay-sdk.md]: Surround code snippet in triple back-ticks

File: docs/source/integrations/ios-fairplay-sdk.md [hunk 23]
```diff
@@ -1,138 +1,140 @@
...
 To cancel a download, call:
- 
+
```
Comment (lorenzo-vualto) [inline docs/source/integrations/ios-fairplay-sdk.md]: Surround code snippet in triple back-ticks

File: docs/source/integrations/ios-fairplay-sdk.md [hunk 24]
```diff
@@ -1,138 +1,140 @@
...
 Legacy methods:
- 
+
 These are the originally provided methods which are for online playback use only, and are provided for ease of upgrade for legacy users. These methods now also require an asset ID in the initialiser to be compatible with the new framework.
- 
-Legacy method 1: Pass in a reference to your AVPlayer instance, the token to be used, and the asset ID when retrieving a license. 
- 
+
+Legacy method 1: Pass in a reference to your AVPlayer instance, the token to be used, and the asset ID when retrieving a license.
+
```
Comment (lorenzo-vualto) [inline docs/source/integrations/ios-fairplay-sdk.md]: Surround code snippet in triple back-ticks

File: docs/source/integrations/ios-fairplay-sdk.md [hunk 25]
```diff
@@ -1,138 +1,140 @@
...
 let player = AVPlayer(url: assetURL!)
 self.drm = vudrmFairPlay(player: player, token: token, assetName: assetName)
-            
-Legacy method 2: Pass in a reference to your stream URL and the token to be used when retrieving a license. An AVAsset will be created by VUDRMFairPlay and this can be used to create your player. 
- 
+  
+Legacy method 2: Pass in a reference to your stream URL and the token to be used when retrieving a license. An AVAsset will be created by VUDRMFairPlay and this can be used to create your player.
+
```
Comment (lorenzo-vualto) [inline docs/source/integrations/ios-fairplay-sdk.md]: Surround code snippet in triple back-ticks

File: docs/source/integrations/ios-fairplay-sdk.md [hunk 26]
```diff
@@ -1,138 +1,140 @@
...
-    		private var token = ""
-   		private var assetName = ""
- 
-● To initialise via view controller on device, populate the URL, Token, and Asset Name fields with valid values. For ease of use, pasting into the URL field in the form URL + space + Token will auto populate both fields correctly. At runtime, any values in the URL and Token fields will override any hard coded values. 
-
-### Limitations ###
+let player = AVPlayer(playerItem: playerItem)
+
+3. Finally, present your player:
+   self.player = player let playerController = AVPlayerViewController() playerController.delegate = self as? AVPlayerViewControllerDelegate self.present(playerController, animated: true, completion: nil) playerController.view.frame = self.view.frame playerController.player = self.player player.play()
```
Comment (lorenzo-vualto) [inline docs/source/integrations/ios-fairplay-sdk.md]: Surround code snippet in triple back-ticks

File: docs/source/integrations/ios-fairplay-sdk.md [hunk 27]
```diff
@@ -1,138 +1,140 @@
...
+   DEMO
+   With the VUDRMFairPlay framework installed, the example demo application can be initialised either via source code in the IDE, or using field entry on device.
+   ● To initialise via code in IDE, before running, simply replace the empty ViewController.swift values with your own valid values.
+   private var assetURL = URL(string: "")
```
Comment (lorenzo-vualto) [inline docs/source/integrations/ios-fairplay-sdk.md]: Surround code snippet in triple back-ticks

File: docs/source/integrations/ios-fairplay-sdk.md [hunk 28]
```diff
@@ -1,138 +1,140 @@
...
-The example includes legacy methods for online playback only. The use of these methods in conjunction with the default method may cause unexpected results and is not recommended. 
-● The SDK does not support playback of protected content when running in the iOS Simulator. Attempting to do so will result in an invalid KEYFORMAT error and the content will not load. 
+The example includes legacy methods for online playback only. The use of these methods in conjunction with the default method may cause unexpected results and is not recommended.
+● The SDK does not support playback of protected content when running in the iOS Simulator. Attempting to do so will result in an invalid KEYFORMAT error and the content will not load.
 
-### Known Issues ###
+### Known Issues
 
-● No known issues at this time. If you believe you have found an issue, please contact us at support@vualto.com 
+● No known issues at this time. If you believe you have found an issue, please contact us at support@vualto.com
```
Comment (lorenzo-vualto) [inline docs/source/integrations/ios-fairplay-sdk.md]: Change to `<support@vualto.com>` to enable as an email link

File: docs/source/integrations/jwplayer.md [hunk 29]
```diff
@@ -0,0 +1,128 @@
+# JWPlayer
+
+[JWPlayer](https://jwplayer.com/) is an industry leading HTML5 video player.
+
+The [vuplay-jwplayer](https://github.com/Vualto/vuplay-jwplayer) repository demonstrates at a lower level how to integrate [vudrm](https://docs.vualto.com/projects/vudrm/en/latest/index.html) with JWPlayer.
+
+## Basic set up
```
Comment (lorenzo-vualto) [inline docs/source/integrations/jwplayer.md]: Change to "setup"

File: docs/source/integrations/integrations.rst (hunk 14 above)
Comment (maniland) [reply docs/source/integrations/integrations.rst]: needed due to weirdness in read the docs making theoplayer disappear

File: docs/source/integrations/ios-fairplay-sdk.md (hunk 27 above)
Comment (maniland) [reply docs/source/integrations/ios-fairplay-sdk.md]: any changes to this document will have to be done by Adam

File: docs/source/integrations/ios-fairplay-sdk.md (hunk 28 above)
Comment (maniland) [reply docs/source/integrations/ios-fairplay-sdk.md]: any changes to this document will have to be done by Adam

File: docs/source/integrations/ios-fairplay-sdk.md (hunk 16 above)
Comment (maniland) [reply docs/source/integrations/ios-fairplay-sdk.md]: any changes to this document will have to be done by Adam

File: docs/source/integrations/ios-fairplay-sdk.md (hunk 18 above)
Comment (maniland) [reply docs/source/integrations/ios-fairplay-sdk.md]: any changes to this document will have to be done by Adam

File: docs/source/integrations/ios-fairplay-sdk.md (hunk 19 above)
Comment (maniland) [reply docs/source/integrations/ios-fairplay-sdk.md]: any changes to this document will have to be done by Adam

File: docs/source/integrations/ios-fairplay-sdk.md (hunk 21 above)
Comment (maniland) [reply docs/source/integrations/ios-fairplay-sdk.md]: any changes to this document will have to be done by Adam

File: docs/source/integrations/ios-fairplay-sdk.md (hunk 22 above)
Comment (maniland) [reply docs/source/integrations/ios-fairplay-sdk.md]: any changes to this document will have to be done by Adam

File: docs/source/integrations/ios-fairplay-sdk.md (hunk 23 above)
Comment (maniland) [reply docs/source/integrations/ios-fairplay-sdk.md]: any changes to this document will have to be done by Adam

File: docs/source/integrations/ios-fairplay-sdk.md (hunk 24 above)
Comment (maniland) [reply docs/source/integrations/ios-fairplay-sdk.md]: any changes to this document will have to be done by Adam

File: docs/source/integrations/ios-fairplay-sdk.md (hunk 25 above)
Comment (maniland) [reply docs/source/integrations/ios-fairplay-sdk.md]: any changes to this document will have to be done by Adam

File: docs/source/integrations/ios-fairplay-sdk.md (hunk 26 above)
Comment (maniland) [reply docs/source/integrations/ios-fairplay-sdk.md]: any changes to this document will have to be done by Adam

File: docs/source/integrations/ios-fairplay-sdk.md (hunk 17 above)
Comment (maniland) [reply docs/source/integrations/ios-fairplay-sdk.md]: any changes to this document will have to be done by Adam

File: docs/source/integrations/ios-fairplay-sdk.md (hunk 20 above)
Comment (maniland) [reply docs/source/integrations/ios-fairplay-sdk.md]: any changes to this document will have to be done by Adam

**Pull request merged**